### ISSUE TITLE
Unconditional expand in grouping table causes more harm than good

### ISSUE THREAD

**Issue by natduca** [catapult-project/catapult#1765]
The unconditional expand rule we have in the grouping table is causing more harm than good. It isn't contextual, as in it expands all subrows with >4 children, which means tables with any real content just get fully expanded. This makes every user of, say, perfinisghts data, have to spend a long while just collapsing the treeview again.

I propose we revert the autoexpand patch, and then find a way to revisit if we have energy. Or, we immediately work on a patch to allow the thing that gets expanded to be defined by a embedder-provided heuristic, e.g. expand the top N interesting cells. But, how to define that api is hard, so may take more time than, say, a revert.

**Comment by natduca** [comment]
Calling @benshayden

**Comment by benshayden** [comment]
I haven't seen anybody spend a while re-collapsing the tree.
Can you demonstrate a sub-optimal situation so that I can see what you're talking about?

NB it only expands subtrees with < 5 children.

I can revert it if you want, but maybe first we could take just a few minutes to see if there might be an easy-enough good-enough fix?
Maybe it should only auto-expand the top 2 or 3 layers?

**Comment by natduca** [comment]
I think the issue here is that I use the tool every day, so its really hitting me hard. Every demo I do gets stalled by this feature. The only real workable solution here is that the "interesting" ones expand --- where interesting is defined by the values being expanded are significant.

I think the ideal here is that the grouping table have a isRowInterestingCallback of form function(row)=>bool ... setter/getter. By default, it returns false. The autoexpansion code uses this callback. If you land just that, then you can re-create the autoexpand in things where you want it to work --- I'd like it to not happen in slice_cost_report for instance, so the default behavior being false is good.

A process note: Though I'd be open to a brainstorm on this, we've got a lot of other places like rail triage etc where I'd prefer to invest our collaboration time, so personally I'd really appreciate a quick n practical solution to this issue.

**Comment by benshayden** [comment]
https://codereview.chromium.org/1454973002

**Comment by natduca** [comment]
In this CL description, there appears to be some confusion about the issue being discussed. The issue was the one that @petrcermak mentioned in codereview when we first landed the original patch , btw. It is that when I open something like the slice cost report, I get this:
![image](https://cloud.githubusercontent.com/assets/412396/11228756/f39daa2a-8d43-11e5-9706-d5a7a1319250.png)

This is because the expand heuristic is expanding things based on how many children they have. Whereas, this is what I would like to see:
![image](https://cloud.githubusercontent.com/assets/412396/11228776/1086e17e-8d44-11e5-969a-904e1c650539.png)

Or, frankly, this:
![image](https://cloud.githubusercontent.com/assets/412396/11228789/2790cc90-8d44-11e5-9cf2-889b42530dcd.png)

Namely, that the most significant items are expanded. Not the ones with lots-of-children-but-with-small-numeric-values.

We can revert, of course, but that doesn't really advance the real issue. I gave that option because that is of course an option if we're swamped... but then we'd need to fix the expansion bug... so ideally we'd find a middle ground.
